imap_processing/codice/codice_l1a_lo_priority.py

```diff
@@ -0,0 +1,256 @@
+                ),
+            ),
+            "spin_sector": xr.DataArray(
+                np.arange(collapse_shape[0], dtype=np.uint8),
+                dims=("spin_sector",),
+                attrs=cdf_attrs.get_variable_attributes(
+                    "spin_sector", check_schema=False
+                ),
+            ),
+            "spin_sector_label": xr.DataArray(
+                np.arange(collapse_shape[0]).astype(str),
+                dims=("spin_sector",),
+                attrs=cdf_attrs.get_variable_attributes(
+                    "spin_sector_label", check_schema=False
+                ),
+            ),
+        },
+        attrs=cdf_attrs.get_global_attributes(logical_source_id),
+    )
+    # Add first few unique variables
+    l1a_dataset["spin_period"] = xr.DataArray(
+        unpacked_dataset["spin_period"].values * constants.SPIN_PERIOD_CONVERSION,
+        dims=("epoch",),
+        attrs=cdf_attrs.get_variable_attributes("spin_period"),
+    )
+    l1a_dataset["k_factor"] = xr.DataArray(
+        np.array([constants.K_FACTOR]),
+        dims=("k_factor",),
+        attrs=cdf_attrs.get_variable_attributes("k_factor_attrs", check_schema=False),
+    )
+    l1a_dataset["voltage_table"] = xr.DataArray(
+        np.array(voltage_data),
+        dims=("esa_step",),
+        attrs=cdf_attrs.get_variable_attributes("voltage_table", check_schema=False),
+    )
+    l1a_dataset["data_quality"] = xr.DataArray(
+        unpacked_dataset["suspect"].values,
+        dims=("epoch",),
+        attrs=cdf_attrs.get_variable_attributes("data_quality"),
+    )
+    l1a_dataset["acquisition_time_per_step"] = xr.DataArray(
+        calculate_acq_time_per_step(sci_lut_data["lo_stepping_tab"]),
+        dims=("esa_step",),
+        attrs=cdf_attrs.get_variable_attributes(
+            "acquisition_time_per_step", check_schema=False
+        ),
+    )
+
+    # Carry over these variables from unpacked data to l1a_dataset
+    l1a_carryover_vars = [
+        "sw_bias_gain_mode",
+        "st_bias_gain_mode",
+        "rgfo_half_spin",
+        "nso_half_spin",
+    ]
+    # Loop through them since we need to set their attrs too
+    for var in l1a_carryover_vars:
+        l1a_dataset[var] = xr.DataArray(
+            unpacked_dataset[var].values,
+            dims=("epoch",),
+            attrs=cdf_attrs.get_variable_attributes(var),
+        )
+
+    # Finally, add species data variables and their uncertainties
+    for idx, species in enumerate(species_names):
+        l1a_dataset[species] = xr.DataArray(
+            species_data[:, idx, :, :],
+            dims=("epoch", "esa_step", "spin_sector"),
+            attrs=cdf_attrs.get_variable_attributes(species),
+        )
+        l1a_dataset[f"unc_{species}"] = xr.DataArray(
+            np.sqrt(species_data[:, idx, :, :]),
+            dims=("epoch", "esa_step", "spin_sector"),
+            attrs=cdf_attrs.get_variable_attributes(species),
+        )
```

Comment:
I'm curious if you can do this instead and avoid the dimensions repeating so that you are using the species xarray data rather than the underlying species numpy array.
```suggestion
        l1a_dataset[f"unc_{species}"] = xr.DataArray(
            np.sqrt(l1a_dataset[species]),
            attrs=cdf_attrs.get_variable_attributes(species),
        )
```